lnwire/custom.go

```diff
@@ -2,13 +2,62 @@ package lnwire
 
 import (
 	"bytes"
-	"errors"
+	"fmt"
 	"io"
 )
 
-// CustomTypeStart is the start of the custom type range for peer messages as
-// defined in BOLT 01.
-var CustomTypeStart MessageType = 32768
+var (
+	// CustomTypeStart is the start of the custom type range for peer
+	// messages as defined in BOLT 01.
+	CustomTypeStart MessageType = 32768
+
+	// CustomTypeOverride contains a set of message types < CustomTypeStart
+	// that lnd allows to be treated as custom messages. This allows us to
+	// override messages reserved for the protocol level and treat them as
+	// custom messages. This set of message types is stored as a global so
+	// that we do not need to pass around state when accounting for this
+	// set of messages in message creation.
+	//
+	// Note: This value should be written only once at startup, then used
+	// as a read-only set of message types.
+	CustomTypeOverride []MessageType
+)
+
+// SetCustomOverrides validates that the set of override types are outside of
+// the custom message range (there's no reason to override messages that are
+// already within the range), and updates the CustomTypeOverride global to hold
+// this set of message types.
+//
+// Note: this function should be called only once at startup/ config parsing
+// to update the global.
+func SetCustomOverrides(overrideTypes []uint64) error {
```

Comment:
We're passing in `uint64` and casting to `MessageType` (`uint16`).  This could result in unexpected truncation (e.g., 65536 becomes 0).

We should probably either pass in `MessageType` or return an error if the type is greater than 65535.